### PR TITLE
Fix ignored warns and time related edge case bug

### PR DESCRIPTION
**Pull request by simonbaird**

### CODE REVIEW

**Comment by simonbaird** [comment]
I'll call it a draft since I think there should be some test coverage added to the warnings fix.

File: policy/lib/main_denies.rego [hunk 1]
```diff
@@ -35,14 +37,27 @@ skip_package(package_name) {
...
+	rule.effective_on
+
+	# The rule is effective in the future but not now
+	time.parse_rfc3339_ns(rule.effective_on) > effective_current_time_ns
+}
+
+# Use the nanosecond epoch defined in the policy config if it is
+# present, otherwise use the real current time
+effective_current_time_ns = now_ns {
+	data.config
```
Comment (simonbaird) [inline policy/lib/main_denies.rego:54]: This could be omitted but being explicit here makes it more understandable I think.

File: policy/release/main.rego [hunk 2]
```diff
@@ -4,6 +4,8 @@ import data.lib
 
 all_denies := lib.current_and_future_denies("release")
 
-deny := lib.current_denies(all_denies)
+all_warns := lib.current_and_future_warns("release")
 
-warn := lib.future_denies(all_denies)
+deny := lib.current_rules(all_denies)
+
+warn := lib.future_rules(all_denies) | lib.current_rules(all_warns)
```
Comment (simonbaird) [inline policy/release/main.rego:11]: A set union.

**Review by zregvart — APPROVED** [review]

**Review by robnester-rh — APPROVED** [review]
LGTM
